packages/amplify_core/lib/src/types/plugin/amplify_datastore_plugin_interface.dart

```diff
@@ -61,6 +61,10 @@ abstract class DataStorePluginInterface extends AmplifyPluginInterface {
     this.authModeStrategy = AuthModeStrategy.defaultStrategy,
   });
 
+  /// Internal use constructor
+  @protected
+  DataStorePluginInterface.emptyConstructor() : this(modelProvider: null);
```

Comment:
Not sure if this is a great name. Open to suggestions here.

Reply:
Curious what's the use case of this constructor?

Reply:
Previously there was a constructor called tokenOnly, which did not accept a modelProvider. The token was the verification token from the `[PlatformformInterface](https://pub.dev/documentation/plugin_platform_interface/latest/plugin_platform_interface/PlatformInterface-class.html)` class. I wasn't sure if we still needed a named constructor, or if the default would be ok.

Reply:
I think we should ditch this in all but DataStore. Even in DataStore, the plugin architecture should be reworked so we don't need this (if I remember it doesn't really do anything anyway), but that can happen later.